File: app/src/main/java/com/michelbarbosa/nytmovies/ui/ListActivity.java
package com.michelbarbosa.nytmovies.ui;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.inputmethod.EditorInfo;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.SearchView;

import com.michelbarbosa.nytmovies.R;

public class ListActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        menu.clear();
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_search, menu);

        MenuItem searchItem = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) searchItem.getActionView();
        searchView.setImeOptions(EditorInfo.IME_ACTION_DONE);
        setApplicationStyle(searchView);

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                setAdapterFilter(newText);
                return false;
            }
        });
        return true;
    }

    private void setApplicationStyle(SearchView searchView) {
        ImageView searchButton = searchView.findViewById(R.id.search_button);
        ImageView closeButton = searchView.findViewById(R.id.search_close_btn);
        TextView textSearch = searchView.findViewById(R.id.search_src_text);
        int color = getResources().getColor(R.color.colorBase);
        searchButton.setColorFilter(color);
        closeButton.setColorFilter(color);
        textSearch.setTextColor(color);
    }

    protected void setAdapterFilter(String newText) {
    }

}

File: app/src/main/java/com/michelbarbosa/nytmovies/ui/ListActivity.kt
package com.michelbarbosa.nytmovies.ui

import android.view.Menu
import android.view.inputmethod.EditorInfo
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.widget.SearchView
import com.michelbarbosa.nytmovies.R

open class ListActivity : BaseActivity() {

    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
        menu.clear()
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_search, menu)
        val searchItem = menu.findItem(R.id.action_search)
        val searchView =
            searchItem.actionView as SearchView
        searchView.imeOptions = EditorInfo.IME_ACTION_DONE
        setApplicationStyle(searchView)
        searchView.setOnQueryTextListener(object :
            SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                findForQuery(query)
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                setAdapterFilter(newText)
                return false
            }
        })
        return true
    }

    private fun setApplicationStyle(searchView: SearchView) {
        val searchButton =
            searchView.findViewById<ImageView>(R.id.search_button)
        val closeButton =
            searchView.findViewById<ImageView>(R.id.search_close_btn)
        val textSearch = searchView.findViewById<TextView>(R.id.search_src_text)
        val color = resources.getColor(R.color.colorBase)
        searchButton.setColorFilter(color)
        closeButton.setColorFilter(color)
        textSearch.setTextColor(color)
    }

    protected open fun setAdapterFilter(newText: String?) {}

    protected open fun findForQuery(query: String) {}

}